docs/src/pages/components/data-grid/components/CustomColumnMenu.js

```diff
@@ -0,0 +1,259 @@
+import { makeStyles } from '@material-ui/core/styles';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import Button from '@material-ui/core/Button';
+import {
+  GridColumnMenu,
+  GridColumnMenuContainer,
+  GridFilterMenuItem,
+  SortGridMenuItems,
+  XGrid,
+} from '@material-ui/x-grid';
+import StarOutlineIcon from '@material-ui/icons/StarOutline';
+
+const useStyles = makeStyles((theme) => ({
+  primary: {
+    background: theme.palette.primary.main,
+    color: theme.palette.primary.contrastText,
+  },
+  secondary: {
+    background: theme.palette.secondary.main,
+    color: theme.palette.secondary.contrastText,
+  },
+}));
+
+function CustomColumnMenuComponent(props) {
+  const classes = useStyles();
+  const { hideMenu, currentColumn, color, ...other } = props;
+
+  if (currentColumn.field === 'name') {
+    return (
+      <GridColumnMenuContainer
+        hideMenu={hideMenu}
+        currentColumn={currentColumn}
+        className={classes[color]}
+        {...other}
+      >
+        <SortGridMenuItems onClick={hideMenu} column={currentColumn} />
+        <GridFilterMenuItem onClick={hideMenu} column={currentColumn} />
+      </GridColumnMenuContainer>
+    );
+  }
+  if (currentColumn.field === 'stars') {
+    return (
+      <GridColumnMenuContainer
+        hideMenu={hideMenu}
+        currentColumn={currentColumn}
+        className={classes[color]}
+        {...other}
+      >
+        <div
+          style={{
+            width: 127,
+            height: 160,
+            display: 'flex',
+            justifyContent: 'center',
+            flexDirection: 'column',
+            alignItems: 'center',
+          }}
+        >
+          <StarOutlineIcon style={{ fontSize: 80 }} />
+        </div>
+      </GridColumnMenuContainer>
+    );
+  }
+  return (
+    <GridColumnMenu
+      hideMenu={hideMenu}
+      currentColumn={currentColumn}
+      className={classes[color]}
+      {...other}
+    />
+  );
+}
+
+CustomColumnMenuComponent.propTypes = {
+  color: PropTypes.string.isRequired,
+  currentColumn: PropTypes.shape({
```

Comment:
too long, see the https://github.com/mui-org/material-ui-x/blob/7215bc63061b0f64471a9adf35dd11fe4b07399d/docs/scripts/formattedTSDemos.js#L87